test/e2e/apimachinery/resource_quota.go

```diff
@@ -1737,6 +1737,12 @@ func newTestReplicaSetForQuota(name, image string, replicas int32) *appsv1.Repli
 
 // newTestServiceForQuota returns a simple service
 func newTestServiceForQuota(name string, serviceType v1.ServiceType, allocateLoadBalancerNodePorts bool) *v1.Service {
```

Comment:
All three places that call this pass `false` for `allocateLoadBalancerNodePorts`.

Can you go thru that and make sure it tests quota for type=NodePort *and* NodePorts allocated in type=LoadBalancer when this is true *and not* NodePorts in type=LoadBalancer when this is false?

Reply:
Not sure if this is what you're expecting, added the happy path for LB type service creation both with and w/o node port, and retain the negative case for resource quota exceed test.

Reply:
Ahh, had a few tries and I just realized that, the `allocateLoadBalancerNodePorts` feature is still Beta and is not enabled in the GA feature only conformance tests (namely `pull-kubernetes-conformance-kind-ga-only-parallel`), so it was failing consistently. That said, in the current resource quota conformance tests, I think we need to assume that the `allocateLoadBalancerNodePorts` feature is not enabled and for each LB type service created, a node port will be allocated. I reverted the new conformance test, and we can enable the `allocateLoadBalancerNodePorts` specific test once the feature is GAed.

Reply:
It worries me that we have more work to do when we GA - it seems likely to get forgotten.  Can we add the code now, but explicitly check the gate for `PreRelease = featuregate.GA`.  That way it starts testing as soon as we mark it GA and when we remove the gate, we will also have to remove that check.

Reply:
Hi @thockin, I think `defaultKubernetesFeatureGates` is not exposed
https://github.com/kubernetes/kubernetes/blob/38f012320fa138260dca09e135ed386026a061bc/pkg/features/kube_features.go#L726
so it's hard to get maturity of a specific feature. By checking feature gate https://github.com/kubernetes/kubernetes/blob/442a69c3bdf6fe8e525b05887e57d89db1e2f3a5/staging/src/k8s.io/component-base/featuregate/feature_gate.go#L86-L96
we can check if a feature is enabled, but we can't get its maturity. So for now the test will be skipped if only GAed features are enabled and will be covered if all beta features are enabled. We have both testbed in our pipeline so both scenarios are covered.

Reply:
conformance tests may not skip under any circumstances.
non-GA features need to be tested in not-a-conformance-test.

Reply:
Followed suggestions from @bentheelder and @liggitt, for now keep the test as an e2e test. Later once we promote the feature to GA, we can promote the test to a conformance test.

Reply:
even if the feature were GA: we usually start by adding a test if missing and then wait for evidence that the test is stable + reliable on some existing CI before promoting it to a test everyone must pass when certifying their clusters 😅

Reply:
Followed suggestion from @aojea, moved the test from e2e to an integration test.